crates/iceberg/src/metadata_scan.rs

```diff
@@ -128,6 +137,135 @@ impl<'a> SnapshotsTable<'a> {
+            Field::new("lower_bound", DataType::Utf8, true),
+            Field::new("upper_bound", DataType::Utf8, true),
+        ]
+    }
+
+    fn schema(&self) -> Schema {
+        Schema::new(vec![
+            Field::new("content", DataType::Int8, false),
+            Field::new("path", DataType::Utf8, false),
+            Field::new("length", DataType::Int64, false),
+            Field::new("partition_spec_id", DataType::Int32, false),
+            Field::new("added_snapshot_id", DataType::Int64, false),
+            Field::new("added_data_files_count", DataType::Int32, false),
+            Field::new("existing_data_files_count", DataType::Int32, false),
+            Field::new("deleted_data_files_count", DataType::Int32, false),
+            Field::new("added_delete_files_count", DataType::Int32, false),
+            Field::new("existing_delete_files_count", DataType::Int32, false),
+            Field::new("deleted_delete_files_count", DataType::Int32, false),
+            Field::new(
+                "partition_summaries",
+                DataType::List(Arc::new(Field::new_struct(
+                    "item",
+                    self.partition_summary_fields(),
+                    false,
+                ))),
+                false,
+            ),
+        ])
+    }
+
+    /// Scans the manifests table.
+    pub async fn scan(&self) -> Result<RecordBatch> {
+        let mut content = PrimitiveBuilder::<Int8Type>::new();
+        let mut path = StringBuilder::new();
+        let mut length = PrimitiveBuilder::<Int64Type>::new();
+        let mut partition_spec_id = PrimitiveBuilder::<Int32Type>::new();
+        let mut added_snapshot_id = PrimitiveBuilder::<Int64Type>::new();
+        let mut added_data_files_count = PrimitiveBuilder::<Int32Type>::new();
+        let mut existing_data_files_count = PrimitiveBuilder::<Int32Type>::new();
+        let mut deleted_data_files_count = PrimitiveBuilder::<Int32Type>::new();
+        let mut added_delete_files_count = PrimitiveBuilder::<Int32Type>::new();
+        let mut existing_delete_files_count = PrimitiveBuilder::<Int32Type>::new();
+        let mut deleted_delete_files_count = PrimitiveBuilder::<Int32Type>::new();
+        let mut partition_summaries = ListBuilder::new(StructBuilder::from_fields(
+            Fields::from(self.partition_summary_fields()),
+            0,
+        ))
+        .with_field(Arc::new(Field::new_struct(
+            "item",
+            self.partition_summary_fields(),
+            false,
+        )));
+
+        if let Some(snapshot) = self.metadata_table.metadata().current_snapshot() {
+            let manifest_list = snapshot
+                .load_manifest_list(
+                    self.metadata_table.0.file_io(),
+                    &self.metadata_table.0.metadata_ref(),
+                )
+                .await?;
+            for manifest in manifest_list.entries() {
+                content.append_value(manifest.content.clone() as i8);
```

Comment:
It's a bit unusual to see something that can use `as u8` but still requires `clone`.